### PR TITLE
feat: Adding sorting algorithm written in java

### PR DESCRIPTION
**Pull request by ziqiang2344**

### CODE REVIEW

File: main.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    public static void swap(String[] array, int c, int j) {
+        String temp = array[c];
+        array[c] = array[j];
+        array[j] = temp;
+    }
+
+    // Initiate last elment as pivot, insert the pivot elment in the array in a sorted manner
+    public static int partition(String[] arr, int left, int right) {
+        // Setting pivot
+        String pivot = arr[left];
```
Comment (Michelle-Lohwt) [inline main.java]: change the pivot to the rightmost element ya, the python version used the rightmost one)

**Review by Michelle-Lohwt — CHANGES_REQUESTED** [review]
Is other files other than `main.java` really needed for this git repo?

**Comment by ziqiang2344** [comment]
> Is other files other than `main.java` really needed for this git repo?

The `.idea` files is auto-generated by the Intellij IDE itself.

`main.class` is generated while compiling the `main.java` file. Can't be deleted as told by the system.

**Comment by Michelle-Lohwt** [comment]
If I copy paste your code to vscode, it runs without the other files, try to avoid other files as the submitted version does not require other files

File: main.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        String pivot = arr[right];
+
+        // Initiate the index of smaller element at rightmost position of pivot
+        int i = (left - 1);
+
+        for (int j = left; j <= right - 1; j++) {
+
+            // If current string is smaller than the pivot
+
+            if (arr[j].compareTo(pivot) < 0) {
```
Comment (Michelle-Lohwt) [inline main.java:29]: why use `compareTo` insetead of `if(arr[j] <= pivot)`

**Review by Michelle-Lohwt — COMMENTED** [review]
Please refer to the comment

File: main.java (hunk 2 above)
Comment (ziqiang2344) [reply main.java:29]: `compareTo` is Java's built-in function inside of `String class` for performing strings comparison. Java doesn't recognize `arr[j] <= pivot` as string operations and hence will show `bad operands` error.

**Review by Michelle-Lohwt — APPROVED** [review]

**Comment by Michelle-Lohwt** [comment]
I think you can put main.class in gitignore

**Pull request merged**